### PR TITLE
POL-178 validate redirect uris

### PR DESCRIPTION
**Pull request by dvoet**

### CODE REVIEW

File: service/src/main/java/bio/terra/externalcreds/services/ProviderService.java [hunk 1]
```diff
@@ -140,6 +142,15 @@ public int refreshExpiringPassports() {
             });
   }
 
+  private void validateRedirectUri(String redirectUri, ProviderProperties providerInfo) {
```
Comment (Shakespeared) [inline service/src/main/java/bio/terra/externalcreds/services/ProviderService.java:145]: What's the reason for filtering this list as opposed to looking for a match?

**Review by Shakespeared — APPROVED** [review]
LGTM just left a question

File: service/src/main/java/bio/terra/externalcreds/services/ProviderService.java (hunk 1 above)
Comment (dvoet) [reply service/src/main/java/bio/terra/externalcreds/services/ProviderService.java:145]: I was being dense. I found `noneMatch`!

**Comment by zarsky-broad** [comment]
Any reason for not using `var` with `ProviderProperties`?

**Review by zarsky-broad — APPROVED** [review]
Code looks good, tests seem comprehensive.

**Review by sarahgibs — APPROVED** [review]
Looks good.

**Pull request merged**